overlord/hookstate/ctlcmd/model.go

```diff
@@ -0,0 +1,187 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/snapcore/snapd/client/clientutil"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/overlord/devicestate"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+)
+
+var (
+	shortModelHelp = i18n.G("Get the active model for this device")
+	longModelHelp  = i18n.G(`
+The model command returns the active model assertion information for this
+device.
+
+By default, the model identification information is presented in a structured,
+yaml-like format, but this can be changed to json by using the --json flag.
+
+Similarly, the active serial assertion can be used for the output instead of the
+model assertion.
+`)
+)
+
+func init() {
+	addCommand("model", shortModelHelp, longModelHelp, func() command { return &modelCommand{} })
+}
+
+type modelCommand struct {
+	baseCommand
+	Assertion bool `long:"assertion"`
+	Json      bool `long:"json"`
+	TermWidth int  `long:"term-width"`
+}
+
+type modelCommandFormatter struct {
+	storeAccount *snap.StoreAccount
+}
+
+// GetEscapedDash must be implemented as a part of the interface
+func (mf modelCommandFormatter) GetEscapedDash() string {
+	return "--"
+}
+
+// LongPublisher must be implemented as a part of the interface
+func (mf modelCommandFormatter) LongPublisher(storeAccountID string) string {
+	if mf.storeAccount == nil {
+		return mf.GetEscapedDash() + "\033[0m"
+	}
+
+	badge := ""
+	if mf.storeAccount.Validation == "verified" {
+		badge = "*"
+	}
+
+	// NOTE this makes e.g. 'Potato' == 'potato', and 'Potato Team' == 'potato-team',
+	// but 'Potato Team' != 'potatoteam', 'Potato Inc.' != 'potato' (in fact 'Potato Inc.' != 'potato-inc')
+	if strings.EqualFold(strings.Replace(mf.storeAccount.Username, "-", " ", -1), mf.storeAccount.DisplayName) {
+		return mf.storeAccount.DisplayName + badge + "\033[0m"
+	}
+	return fmt.Sprintf("%s (%s%s%s)", mf.storeAccount.DisplayName, mf.storeAccount.Username, badge, "\033[0m")
+}
+
+func newTabWriter(output io.Writer) *tabwriter.Writer {
+	minWidth := 2
+	tabWidth := 2
+	padding := 2
+	padchar := byte(' ')
+	return tabwriter.NewWriter(output, minWidth, tabWidth, padding, padchar, 0)
+}
+
+// reportError prints the error message to stderr
+func (c *modelCommand) reportError(format string, a ...interface{}) {
+	w := newTabWriter(c.stderr)
+	fmt.Fprintf(w, format, a...)
+	w.Flush()
+}
+
+// checkGadgetOrModel verifies that the current snap context is either a gadget snap or that
+// the snap shares the same publisher as the current model assertion.
+func (c *modelCommand) checkGadgetOrModel(st *state.State, snapName string) error {
```

Comment:
this is missing the snapd-control bit of the spec?

Reply:
Yea totally missed it - I have added the functionality now, and would love another check from you on this